HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,52 @@
+# References:
+# http://hadooptutorial.wikispaces.com/Iterative+MapReduce+and+Counters
+# http://www.slideshare.net/jhammerb/lec5-pagerank
+# The CSV and Spark manual pages
+
+INF = 999999999
+
+# Map: emit neighbour nodes if GRAY, otherwise
+# just emit the same thing if WHITE or BLACK
+def processNode((k,v)):
+    curDist, adj, col = v
+    ret = []
+    if col == 1: # Emit all neighbouring nodes and cur
+        for nb in adj:
```

Comment:
You should have avoided doing this in python, but use spark's functions (such as join, subtractByKey and union) instead.